src/sentry/discover/endpoints/discover_key_transactions.py

```diff
@@ -0,0 +1,119 @@
+            if serializer.is_valid():
+                data = serializer.validated_data
+                base_filter["transaction"] = data["transaction"]
+
+                KeyTransaction.objects.create(**base_filter)
+                return Response(status=201)
+            else:
+                return Response(serializer.errors, status=400)
+
+    def get(self, request, organization):
+        """ Get the paged Key Transactions for a user """
+        if not self.has_feature(request, organization):
+            return self.response(status=404)
+
+        params = self.get_filter_params(request, organization)
+        fields = request.GET.getlist("field")[:]
+        orderby = self.get_orderby(request)
+
+        queryset = KeyTransaction.objects.filter(organization=organization, owner=request.user)
+
+        results = query(
+            fields,
+            None,
+            params,
+            orderby=orderby,
+            referrer="discover.key_transactions",
+            # The snuba query for transactions is of the form
+            # (transaction="1" AND project=1) OR (transaction="2" and project=2) ...
+            # which the schema intentionally doesn't support so we cannot do an AND in OR
+            # so here the "and" operator is being instead to do an AND in OR query
+            conditions=[
+                [
+                    # First layer is Ands
+                    [
+                        # Second layer is Ors
+                        [
+                            "and",
+                            [
+                                [
+                                    "equals",
+                                    # Without the outer ' here, the transaction will be treated as another column
+                                    # instead of a string. This isn't an injection risk since snuba is smart enough to
+                                    # handle escaping for us.
+                                    ["transaction", u"'{}'".format(transaction.transaction)],
+                                ],
+                                ["equals", ["project_id", transaction.project.id]],
+                            ],
+                        ],
+                        "=",
+                        1,
+                    ]
+                    for transaction in queryset
+                ]
+            ],
+        )
+
+        return Response(
+            self.handle_results_with_meta(request, organization, params["project_id"], results),
+            status=200,
+        )
+
+    def delete(self, request, organization):
+        """ Remove a Key transaction for a user """
+        if not self.has_feature(request, organization):
+            return self.response(status=404)
+
+        projects = self.get_projects(request, organization)
+
+        if len(projects) != 1:
+            return Response({"detail": "Only 1 project per Key Transaction"}, status=400)
+        project = projects[0]
```

Comment:
You could extract this section out into a method as you have in repeated a few times.